core/src/main/java/com/linecorp/armeria/client/endpoint/healthcheck/AbstractHealthCheckedEndpointGroupBuilder.java

```diff
@@ -45,6 +49,9 @@
     private ClientFactory clientFactory = ClientFactory.ofDefault();
     private Function<? super ClientOptionsBuilder, ClientOptionsBuilder> configurator = Function.identity();
     private int port;
+    private HealthCheckStrategy healthCheckStrategy = HealthCheckStrategy.all();
```

Comment:
I think we can remove this field and create a new strategy only in `build()` when `partialHealthCheckStrategyBuilder` is `null`.

Reply:
fixed